Source/objects.cpp

```diff
@@ -5195,8 +5180,11 @@ void ObjChangeMap(int x1, int y1, int x2, int y2)
 		ObjL2Special(2 * x1 + 16, 2 * y1 + 16, 2 * x2 + 17, 2 * y2 + 17);
 		AddL2Objs(2 * x1 + 16, 2 * y1 + 16, 2 * x2 + 17, 2 * y2 + 17);
 	}
+	if (leveltype == DTYPE_CAVES) {
+		AddL3Objs(2 * x1 + 16, 2 * y1 + 16, 2 * x2 + 17, 2 * y2 + 17);
+	}
 	if (leveltype == DTYPE_CRYPT) {
-		AddL5Objs(2 * x1 + 16, 2 * y1 + 16, 2 * x2 + 17, 2 * y2 + 17);
+		AddCryptObjects(2 * x1 + 16, 2 * y1 + 16, 2 * x2 + 17, 2 * y2 + 17);

```

Comment:
Old `AddL5Objs` and `AddCryptObjects` aren't identically (checks different piece IDs).
Is this no problem, cause the method was never executed or in hellfire the piece's aren't used?
(Perhaps I miss something, I'm no map expert 😉)

Reply:
`AddL5Objs` was a very rudimentary split from `AddL5Objs`, `AddCryptObjects` was a fully "optimized" version. Both where added by me last week when splitting crypt logic from cathedral, but at two different points of the call chain.